File: deno/src/base/usr/usr.resolver.ts
import {
  useContext,
} from "/lib/context.ts";

import type {
  MutationLoginArgs,
  Mutation,
  ChangePasswordInput,
} from "/gen/types.ts"

/**
 * 登录获得 authorization
 * @param {MutationLoginArgs["input"]} input 登录信息
 *  username 用户名
 *  password 密码,传递进来的密码已经被前端md5加密过一次
 *  tenant_id 租户id
 *  org_id 组织id
 *  lang 语言编码
 */
export async function login(
  input: MutationLoginArgs["input"],
): Promise<Mutation["login"]> {
  const {
    login,
  } = await import("./usr.service.ts");
  const context = useContext();
  
  context.notVerifyToken = true;
  context.is_tran = true;
  return await login(input);
}

export async function getLoginInfo() {
  const {
    getLoginInfo,
  } = await import("./usr.service.ts");
  return await getLoginInfo();
}

export async function selectLang(lang: string) {
  const {
    selectLang,
  } = await import("./usr.service.ts");
  return await selectLang(lang);
}

// 修改密码
export async function changePassword(
  input: ChangePasswordInput,
) {
  const {
    changePassword,
  } = await import("./usr.service.ts");
  return await changePassword(input);
}

File: rust/src/src/base/usr/usr_resolver.rs
use anyhow::Result;

use crate::common::context::Ctx;

use super::usr_service;

use super::usr_model::{
  LoginInput,
  Login,
  GetLoginInfo,
  ChangePasswordInput,
};

/// 登录, 获得token
pub async fn login(
  ctx: &Ctx,
  input: LoginInput,
) -> Result<Login> {
  
  let res = usr_service::login(
    ctx,
    input,
  ).await?;
  
  Ok(res)
}

/// 选择语言
pub async fn select_lang(
  ctx: &mut Ctx,
  lang: String,
) -> Result<String> {
  
  let res = usr_service::select_lang(
    ctx,
    lang,
  ).await?;
  
  Ok(res)
}

/// 修改密码
pub async fn change_password(
  ctx: &Ctx,
  input: ChangePasswordInput,
) -> Result<bool> {
  
  let res = usr_service::change_password(
    ctx,
    input,
  ).await?;
  
  Ok(res)
}

pub async fn get_login_info(
  ctx: &Ctx,
) -> Result<GetLoginInfo> {
  
  let res = usr_service::get_login_info(
    ctx,
  ).await?;
  
  Ok(res)
}
